Compilation error since today morning with cudnn 5.1.10

------------------------

### System information
Memory : 16GB
Processor: Intel® Core™ i7-7700HQ CPU @ 2.80GHz × 8 
GPU: GeForce GTX 1060/PCIe/SSE2
OS Type- 64 bit

- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: 16.04
- **TensorFlow installed from (source or binary)**: Source
- **TensorFlow version (use command below)**: 1.3
- **Python version**: 2.7
- **Bazel version (if compiling from source)**: 0.5.4
- **CUDA/cuDNN version**: CUDA 8.0 CUDNN 5.1.10
- **GPU model and memory**: GTX 1060 (6GB)
- **Exact command to reproduce**: bazel build --config=opt --config=cuda //tensorflow/tools/pip_package:build_pip_package


### Describe the problem
Since today morning, I've been getting this error in compiling TF with cudnn 5.1.10 where it seems to be looking for a v6 file even though I've configured it to use 5.1.10. I do not see the same error when building with cudnn6.0.  I compiled it last night with the same exact configuration and it was compiling fine. I've even wiped my system clean and reinstalled the whole environment to make sure there weren't any corrupted libraries. But the error still persists.

### Source code / logs
ERROR: /tensorflow/tensorflow/stream_executor/BUILD:52:1: C++ compilation of rule '//tensorflow/stream_executor:cuda_platform' failed (Exit 1).
tensorflow/stream_executor/cuda/cuda_dnn.cc: In member function 'cudnnStatus_t perftools::gputools::cuda::wrap::WrapperShim__cudnnSetRNNDescriptor_v6::operator()(perftools::gputools::cuda::CUDAExecutor*, Args ...)':
tensorflow/stream_executor/cuda/cuda_dnn.cc:140:30: error: '::cudnnSetRNNDescriptor_v6' has not been declared
       cudnnStatus_t retval = ::__name(args...);                    \
                              ^
tensorflow/stream_executor/cuda/cuda_dnn.cc:235:3: note: in expansion of macro 'PERFTOOLS_GPUTOOLS_CUDNN_WRAP'
   __macro(cudnnSetRNNDescriptor_v6)                           \
   ^
tensorflow/stream_executor/cuda/cuda_dnn.cc:240:1: note: in expansion of macro 'CUDNN_DNN_ROUTINE_EACH_R5'
 CUDNN_DNN_ROUTINE_EACH_R5(PERFTOOLS_GPUTOOLS_CUDNN_WRAP)
 ^
tensorflow/stream_executor/cuda/cuda_dnn.cc: In function 'int perftools::gputools::cuda::{anonymous}::CudnnDataTypeToByteSize(cudnnDataType_t)':
tensorflow/stream_executor/cuda/cuda_dnn.cc:902:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
tensorflow/stream_executor/cuda/cuda_dnn.cc: In member function 'int perftools::gputools::cuda::CudnnRnnParamsDescriptor::GetRegionCountPerLayer() const':
tensorflow/stream_executor/cuda/cuda_dnn.cc:1255:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
tensorflow/stream_executor/cuda/cuda_dnn.cc: In function 'cudnnRNNInputMode_t perftools::gputools::cuda::{anonymous}::ToCudnnRnnInputMode(perftools::gputools::dnn::RnnInputMode)':
tensorflow/stream_executor/cuda/cuda_dnn.cc:865:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
tensorflow/stream_executor/cuda/cuda_dnn.cc: In function 'cudnnDirectionMode_t perftools::gputools::cuda::{anonymous}::ToCudnnRnnDirectionMode(perftools::gputools::dnn::RnnDirectionMode)':
tensorflow/stream_executor/cuda/cuda_dnn.cc:877:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
tensorflow/stream_executor/cuda/cuda_dnn.cc: In function 'cudnnRNNMode_t perftools::gputools::cuda::{anonymous}::ToCudnnRnnMode(perftools::gputools::dnn::RnnMode)':
tensorflow/stream_executor/cuda/cuda_dnn.cc:889:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
tensorflow/stream_executor/cuda/cuda_dnn.cc: In function 'cudnnDataType_t perftools::gputools::cuda::{anonymous}::ToCudnnDataType(perftools::gputools::dnn::DataType, perftools::gputools::dnn::DataLayout)':
tensorflow/stream_executor/cuda/cuda_dnn.cc:853:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
tensorflow/stream_executor/cuda/cuda_dnn.cc: In function 'cudnnConvolutionFwdAlgo_t perftools::gputools::cuda::{anonymous}::ToConvForwardAlgo(perftools::gputools::dnn::AlgorithmDesc)':
tensorflow/stream_executor/cuda/cuda_dnn.cc:297:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
tensorflow/stream_executor/cuda/cuda_dnn.cc: In function 'cudnnConvolutionBwdDataAlgo_t perftools::gputools::cuda::{anonymous}::ToConvBackwardDataAlgo(perftools::gputools::dnn::AlgorithmDesc)':
tensorflow/stream_executor/cuda/cuda_dnn.cc:320:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
tensorflow/stream_executor/cuda/cuda_dnn.cc: In function 'cudnnConvolutionBwdFilterAlgo_t perftools::gputools::cuda::{anonymous}::ToConvBackwardFilterAlgo(perftools::gputools::dnn::AlgorithmDesc)':
tensorflow/stream_executor/cuda/cuda_dnn.cc:342:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
tensorflow/stream_executor/cuda/cuda_dnn.cc: At global scope:
tensorflow/stream_executor/cuda/cuda_dnn.cc:560:13: warning: 'bool perftools::gputools::cuda::TensorOpMathEnabled()' defined but not used [-Wunused-function]
 static bool TensorOpMathEnabled() {
             ^
tensorflow/stream_executor/cuda/cuda_dnn.cc:129:26: warning: 'tensorflow::thread::ThreadPool* perftools::gputools::cuda::wrap::GetCudaThreadpool()' defined but not used [-Wunused-function]
 static port::ThreadPool* GetCudaThreadpool() {
                          ^
tensorflow/stream_executor/cuda/cuda_dnn.cc:2001:20: warning: 'perftools::gputools::dnn::AlgorithmDesc perftools::gputools::cuda::{anonymous}::GetCudnnConvolutionForwardAlgorithm(perftools::gputools::Stream*, perftools::gputools::cuda::CUDAExecutor*, void*, int, const perftools::gputools::dnn::AlgorithmConfig&, bool, const perftools::gputools::cuda::ScopedTensorDescriptor&, const perftools::gputools::cuda::ScopedFilterDescriptor&, const perftools::gputools::cuda::ScopedConvolutionDescriptor&, const perftools::gputools::cuda::ScopedTensorDescriptor&, perftools::gputools::ScratchAllocator*, perftools::gputools::DeviceMemory<unsigned char>*)' defined but not used [-Wunused-function]
 dnn::AlgorithmDesc GetCudnnConvolutionForwardAlgorithm(
                    ^
Target //tensorflow:libtensorflow_all.so failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 666.701s, Critical Path: 125.43s

--- a/tensorflow/stream_executor/cuda/cuda_dnn.cc
+++ b/tensorflow/stream_executor/cuda/cuda_dnn.cc
@@ -232,7 +232,6 @@ CUDNN_DNN_ROUTINE_EACH_R3(PERFTOOLS_GPUTOOLS_CUDNN_WRAP)
   __macro(cudnnRNNBackwardData)                               \
   __macro(cudnnRNNBackwardWeights)                            \
   __macro(cudnnSetRNNDescriptor)                              \
-  __macro(cudnnSetRNNDescriptor_v6)                           \
   __macro(cudnnGetFilterNdDescriptor)
 
 // clang-format on
@@ -245,7 +244,8 @@ CUDNN_DNN_ROUTINE_EACH_R5(PERFTOOLS_GPUTOOLS_CUDNN_WRAP)
 // clang-format off
 #if CUDNN_VERSION >= 6000
 #define CUDNN_DNN_ROUTINE_EACH_R6(__macro)                    \
-  __macro(cudnnConvolutionBiasActivationForward)
+  __macro(cudnnConvolutionBiasActivationForward)              \
+  __macro(cudnnSetRNNDescriptor_v6)
 
 // clang-format on
 CUDNN_DNN_ROUTINE_EACH_R6(PERFTOOLS_GPUTOOLS_CUDNN_WRAP)

